### PR TITLE
expose SlaveComputer#taskListener to subclasses

### PR DESCRIPTION
**Pull request by ndeloof**
@reviewbybees

### CODE REVIEW

File: core/src/main/java/hudson/slaves/SlaveComputer.java [hunk 1]
```diff
@@ -189,6 +189,10 @@ public Slave getNode() {
         }
     }
 
+    protected TaskListener gettaskListener() {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/slaves/SlaveComputer.java]: getTaskListener() . Actually I would prefer a more explicit method name (e.g. getSlaveLog() or whatever). Could be also fixed by Javadoc.

Comment (KostyaSha) [inline core/src/main/java/hudson/slaves/SlaveComputer.java]: Why protected?

**Comment by ndeloof** [comment]
so it is visible to subclasses, maybe also make sense to be public, I don't have a use-case for it in mind.

**Comment by jglick** [comment]
Use case? Downstream PR?

**Comment by ndeloof** [comment]
use case is a hack I made with Yoann, for which we found a workaround, but considered this would have been useful if this were available, so try to make jenkins API a bit better (sic)

File: core/src/main/java/hudson/slaves/SlaveComputer.java (hunk 1 above)
Comment (stephenc) [inline core/src/main/java/hudson/slaves/SlaveComputer.java]: `s/gettaskListener/getTaskListener/g`

**Comment by stephenc** [comment]
:bug: for not following the JavaBeans naming rules

**Comment by jglick** [comment]
> use case is a hack I made with Yoann, for which we found a workaround, but considered this would have been useful if this were available

So…where is the PR showing how this would circumvent the need for that workaround? If the code in question is proprietary, I would expect at least a detailed description of the situation, if not a mockup.

**Comment by jglick** [comment]
(and a proprietary PR to match)

**Comment by jglick** [comment]
Closed why?

**Comment by oleg-nenashev** [comment]
@ndeloof I would like to delete the branch. Are you fine with that?

**Comment by ndeloof** [comment]
Sure

**Pull request closed without merging**